operations/boot_orchestration/Session_Templates.md

```diff
@@ -94,6 +94,63 @@ Each boot set also specifies a set of nodes to be applied to.  There are three d
 
     Consult the `cray-hms-base-config` Kubernetes ConfigMap in the `services` namespace for a listing of the available roles and sub-roles on the system.
 
+#### Rootfs Providers
+
+  The 'rootfs' is the root file system.
+  
+  `rootfs_provider` identifies the mechanism that providers the root file system for the node. 
+
+  In the case of the Cray Operating System (COS) image, the rootfs provider is HPE’s Content Projection Service (CPS), which uses HPE’s Data Virtualization Service (DVS) to deliver the content. CPS projects the root file system onto the nodes as a squashfs image. This is provided via an overlay file system which is set up in dracut. 
+
+
+  `rootfs_provider_passthrough` is a string that is passed through to the provider of the rootfs. This string can contain additional information that the provider will act upon. 
+
+  Both the `rootfs_provider` and `rootfs_provider_passthrough` parameters are used to construct the value of the kernel boot parameter `root` that BOS sends to the node.
+
+  BOS constructs the kernel boot parameter `root` per the following syntax.
+  ```bash
+  root=<Protocol>:<Root FS location>:<Etag>:<RootFS-provider-passthrough parameters>
+  ```
+
+  BOS fills in the protocol based off of the value provided in `rootfs_provider`. If BOS does not know the rootfs_provider, it omits the Protocol field. Currently, BOS only recognizes the rootfs_provider ‘cpss3’.
+  BOS finds the ‘Root FS provider’ and ‘Etag’ values in the manifest file in the session template in the [boot set](#boot-artifacts-and-s3).
+  The rootfs_provider_passthrough parameters are appended to the `root` parameter without modification. They are ‘passed through’ as the name imples.
+
+  `root` kernel parameter example:
+
+  ```bash
+  root=craycps-s3:s3://boot-images/b9caaf66-c0b4-4231-aba7-a45f6282b21d/rootfs:f040d70bd6fabaf91838fe4e484563cf-211:dvs:api-gw-service-nmn.local:300:nmn0 nmd_data=url=s3://boot-images/b9caaf66-c0b4-4231-aba7-a45f6282b21d/rootfs,etag=f040d70bd6fabaf91838fe4e484563cf-211
+  ```
+
+  The following table dissects and explains the different pieces in the preceding example.
+  
+  **Table:** kernel boot parameter `root` syntax
+  |Field|Example Value|Explanation|
+  |-----|-------------|-----------|
+  |Protocol|craycps-s3|The protocol used to mount the root file system, using CPS|
```

Comment:
I wasn't clear when I first read this if this explanation field always refers to a CPS protocol or not.  Is that true, or is this only using CPS because it's in the protocol name?

Reply:
I attempted to make this clearer. Let me know if I succeeded.